SQLite documentation (#604)
fermyon/developer: content/spin/dynamic-configuration.md

```

You must individually grant each component access to the stores that it needs to use. To do this, use the `component.key_value_stores` entry in the component manifest within `spin.toml`. See [Spin Key Value Store](kv-store-api-guide.md) for more details. 

### SQLite Storage Runtime Configuration

> Spin provides built-in SQLite storage. By default, this is backed by a database that Spin creates for you. However, you can use the Spin runtime configuration file (`runtime-config.toml`) to add and customize local SQLite databases.

Reviewer: ```suggestion
Spin provides built-in SQLite storage. By default, this is backed by a database that Spin creates for you. However, you can use the Spin runtime configuration file (`runtime-config.toml`) to point to other SQLite databases.
```

Runtime config will eventually support non-local SQLite databases. We could wait until we actually have that support, but I think it would also be fine to just remove the "local" qualifier.

Author: I retained "add and customise" to hint that you can both add new entries and modify the default one, but have removed "local."

> The following example `runtime-config.toml` tells Spin to map the `default` database to a SQLite database elsewhere in the file system:

Author: team "No it should rhyme with 'granite' not with 'light'" has entered the arena 

![image](https://github.com/fermyon/developer/assets/865538/ecbf35d3-9d7d-4e78-91ec-1169988250bf)

> The following example `runtime-config.toml` tells Spin to map the `default` database to a SQLite database elsewhere in the file system:

Author: anyway changed it to "an SQLite database" throughout, and thanking providence that English doesn't modify the subsequent word based on the ending

```toml
[sqlite_database.default]
path = "/planning/todo.db"
```

> If you need more than one database, you can configure multiple stores, each with their own name:

Reviewer 2: ```suggestion
If you need more than one database, you can configure multiple stores, each with its name:
```

Reviewer 2: "each with its own name" is fine; it makes enough sense. 
Initially, I was coming from the point that the "own" name is redundant because an entity's name is that entity's own name by default. However, I do apologise if we start to split hairs; not my intention. Sometimes things just jump out, and I am compelled to tweak.

Author: Well, I might possibly be able to identify with that...!

# This defines a new store named finance
[sqlite_database.finance]
path = "/super/secret/monies.db"
```

Spin creates any database files that don't exist.  However, it is up to you to delete them when you no longer need them.

Spin can also use [libSQL](https://libsql.org/) databases accessed over HTTPS.  libSQL is fully compatible with SQLite but provides additional features including remote, distributed databases.

> Spin does not provide libSQL access to file-based databases, only databases served over HTTPS.

To use libSQL, set `type = "libsql"` in your `runtime-config.toml` entry.  You must then provide a `url` and authentication `token` instead of a file path.  For example, this entry tells Spin to map the `default` database to a libSQL service running on `libsql.example.com`:

```toml
# This tells Spin to use the remote host as its default database
[sqlite_database.default]
type = "libsql"
url = "sensational-penguin-ahacker.libsql.example.com"
token = "a secret"
```

Spin does _not_ create libSQL databases.  Use your hosting service's tools, or the `libsql` command line, to create them.  You can still set up tables and data in a libSQL database via `spin up --sqlite`.

> > Some libSQL service documentation shows URLs with a `libsql://` scheme.  Don't include a scheme in the `runtime-config.toml` `url` field.  Include only the host name.

Author: I thought about this at the time, and went back and forth because I wasn't sure about forward compatibility risks.  But I agree we should consider it.  I've raised an issue: https://github.com/fermyon/spin/issues/1635

The `default` database will still be defined, even if you add other databases.

> By default, components will not have access to any of these databases (even the default one). You must grant each component access to the databases that it needs to use. To do this, use the `component.sqlite_databases` entry in the component manifest within `spin.toml`. See [SQLite Database](/spin/sqlite-api-guide.md) for more details. 

Reviewer 3: I'm not sure if this is the right article, but we should instruct users on how to delete their DB if they so desire.  Perhaps that's just adding a few sentences informing them they'll have to manually delete the directory.

Author: I've added a sentence about custom databases, which is what this page covers.  But were you referring to the default database?
